### PR TITLE
First pass at semi-automated conversion of WebKit reftests submitted …

### PR DESCRIPTION
**Pull request by hillbrad**
…by Blink as part of https://github.com/w3c/web-platform-tests/pull/1844

### CODE REVIEW

**Comment by sideshowbarker** [comment]
Please see the lint-tool error report at https://travis-ci.org/w3c/web-platform-tests/builds/68251248 and the docs at https://github.com/w3c/web-platform-tests/blob/master/docs/lint-tool.md#fixing-lint-errors on how to fix particular errors reported by the lint tool (including `W3C-TEST.ORG` errors).

File: content-security-policy/blink-contrib-2/base-uri-allow.sub.html.sub.headers [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Expires: Mon, 26 Jul 1997 05:00:00 GMT
+Cache-Control: no-store, no-cache, must-revalidate
+Cache-Control: post-check=0, pre-check=0, false
+Pragma: no-cache
+Set-Cookie: base-uri-allow={{$id:uuid()}}; Path=/content-security-policy/blink-contrib-2
+Content-Security-Policy: base-uri http://w3c-test.org; script-src 'self' 'unsafe-inline' http://w3c-test.org; connect-src 'self'; report-uri /content-security-policy/support/report.py?op=put&reportID={{$id}}
```
Comment (kristijanburnik) [inline content-security-policy/blink-contrib-2/base-uri-allow.sub.html.sub.headers]: Will the base-uri present an issue when ran locally?
E.g. over http://web-platform.test:8000 ?

Comment (hillbrad) [inline content-security-policy/blink-contrib-2/base-uri-allow.sub.html.sub.headers]: I developed and debugged this (base-uri) test locally, and deliberately
picked stable resources on w3c-test.org to re-base against, so I believe it
does work.  (this test required a good deal of manual work)

Comment (hillbrad) [inline content-security-policy/blink-contrib-2/base-uri-allow.sub.html.sub.headers]: I'll give it a try with "www1" instead and see if I can make it work and
not fail lint.

Comment (hillbrad) [inline content-security-policy/blink-contrib-2/base-uri-allow.sub.html.sub.headers]: Was broken, fixed now.

**Comment by sideshowbarker** [comment]
These tests have already gone through upstream blink-project review, right? (To be clear, I mean the substance of the tests, not the changes you made to make them runnable under the wpt infrastructure.)

If they have in fact been through upstream browser-project review, then the only review we need to do here is to confirm that they run as expected under the wpt infrastructure, and the changes done to make them do that follow wpt best-practices and are otherwise sane.

**Comment by hillbrad** [comment]
Sounds right to me.  A few have been modified by me to reflect current spec
behavior which has changed since they were written.  But on the whole I
think reviewing a small sample for best practices would be appropriate.

**Comment by sideshowbarker** [comment]
General comment: For any of the tests that show users text saying “this test passes if”, if users have to actually do some manual check of some kind (e.g., checking the console), then the test file should be renamed with `-manual`. If instead the user doesn’t need to actually do any manual check (because the harness is automatically doing the check and knows whether it passed or failed), then the sentence “this test passes if” should be removed (not shown to users).

**Comment by hillbrad** [comment]
I'll do some regex work on the descriptions.  I've confirmed that all the
tests checked in with this bath run in an automated fashion.  The WebKit
reftest framework relied on console messages; I've converted that to rely
on CSP reporting, and shimmed other uses of console.log and alert() to the
local "log" and "alert_assert" functions.

**Comment by hillbrad** [comment]
Also the form tests can be run in an automated fashion, the narrative
description is just to indicate that the popup warning is expected.

**Comment by sideshowbarker** [comment]
@hillbrad Is this ready to be merged? Or are you still waiting to get review from anybody else in particular?

**Comment by jgraham** [comment]
Mind squashing this and rewriting the commit message (presumably this is no longer a "first pass" if it's going to land).

**Comment by hillbrad** [comment]
Done.

**Comment by jgraham** [comment]
Sorry, I hadn't looked at this before, but please make a followup that removes the use of console.*

**Comment by sideshowbarker** [comment]
> Sorry, I hadn't looked at this before, but please make a followup that removes the use of console.*

@jgraham My bad. I reviewed the tests but I wasn’t aware we had a policy about avoiding `console.*`? What’s the rationale for not using it? I think I probably know what your answer is but I’d like to confirm because I think we should add some info about this to the docs.

Actually, along with updating the docs, shouldn’t we also have the lint tool check for this? If so, I’ll make a patch for the lint tool and open a PR.

Note also that we have around 10 or so existing tests in the repo that already use `console.*`. Should we change those as well?

**Comment by jgraham** [comment]
The rationale is that it's non-standard, almost never the feature actually being tested, and pure overhead when running in automation. It's almost always used in a way that's like leaving extra `printf` in production code i.e. as leftover debugging that isn't actually useful to the next person running the test. I think that in WebKit/Blink tests it can be used as part of the actual test output, but that isn't how reporting in testharness.js tests works, so the idiom doesn't carry over.

**Pull request merged**